File: ex5_factorial/src/Factorial.java
import java.math.BigInteger;

class Factorial {
    public static BigInteger factHelper(BigInteger n, BigInteger acc) {
        if (n.equals(BigInteger.ZERO))
            return acc;
        else
            return factHelper(n.subtract(BigInteger.ONE), acc.multiply(n));
    }

    public static BigInteger fact(int n) {
        return factHelper(BigInteger.valueOf(n), BigInteger.ONE);
    }

    public static void main(String[] args) {
        System.out.println(fact(100000));
    }
}

File: ex5_factorial/src/Factorial.cs
using System;
using System.Numerics;

public class Factorial
{
    private static BigInteger FactHelper(BigInteger n, BigInteger acc) {
        if (n.IsZero)
            return acc;
        else
            return FactHelper(n - BigInteger.One, acc * n);
    }

    public static BigInteger Fact(int n) {
        return FactHelper(new BigInteger(n), BigInteger.One);
    }

    public static void Main()
    {
      Console.WriteLine(Fact(10000));
    }
}
